inject/inject/src/main/java/io/helidon/inject/RegistryServiceProvider.java

```diff
@@ -21,16 +21,22 @@
 import java.util.Set;
 
 import io.helidon.common.types.TypeName;
+import io.helidon.inject.service.InjectionPointProvider;
 import io.helidon.inject.service.Ip;
 import io.helidon.inject.service.Qualifier;
 import io.helidon.inject.service.ServiceInfo;
+import io.helidon.inject.service.ServiceProvider;
 
 /**
  * Provides management lifecycle around services.
+ * <p>
+ * This should never be implemented by user code. The only exception is extensibility of the service registry itself.
```

Comment:
This split between `ServiceProvider` and `RegistryServiceProvider` seems completely unnatural to me.

If it shouldn't be implemented then make it a (package private) class, and use JPMS system to protect its use. And it should be called Internal or something.